### DISCUSSION TITLE
Usage with Google Analytics API

### DISCUSSION THREAD

**Discussion by ekofi** [nextauthjs/next-auth#4104] [Help] [answered]
### Question 💬

Hi, can I use next-auth with Analytics API? My scope is 
**Google Analytics API | .../auth/analytics.readonly | See and download your Google Analytics data**

And then I need to get Reporting API v4 (I need to access my customers' Analytics data)

If yes, how can I do that? Thanks

### How to reproduce ☕️

I added the scope

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by ekofi** [comment, marked as answer]
```
import NextAuth from "next-auth";
import GoogleProvider from "next-auth/providers/google";

export default NextAuth({
  // Configure one or more authentication providers
  providers: [
    GoogleProvider({
      clientId: process.env.GOOGLE_CLIENT_ID,
      clientSecret: process.env.GOOGLE_CLIENT_SECRET,
      authorizationUrl:
        "https://accounts.google.com/o/oauth2/v2/auth?prompt=consent&access_type=offline&response_type=code",
      scope:
        "https://www.googleapis.com/auth/userinfo.profile https://www.googleapis.com/auth/userinfo.email https://www.googleapis.com/auth/analytics.readonly",
    }),
  ],
...
});

```

**Comment by MatthieuVeillon** [comment]
Hi @ekofi,

I have a similar issue, in the end how did you end up fetching the Google Analytics API ? by passing a token directly to the API call or configuring a google client SDK ?

I have the following nextAuth config : 

```
export default NextAuth({
  adapter: PrismaAdapter(prisma),
  providers: [
    GoogleProvider({
      authorization: {
        params: {
          prompt: "consent",
          access_type: "offline",
          response_type: "code",
          scope:
            "https://www.googleapis.com/auth/userinfo.profile https://www.googleapis.com/auth/userinfo.email https://www.googleapis.com/auth/analytics.readonly",
        },
      },
      clientId: process.env.GOOGLE_CLIENT_ID,
      clientSecret: process.env.GOOGLE_CLIENT_SECRET,
    }),
  ],
});
```


which seems to work fine, but I'm not sure how to pass the access token to any following request as I'm using a database strategy.

**Reply by ekofi** [reply]
I just did directly API call. You can use fetch & axios and put the token variable in. Thanks
